integration/simulation/network/geth_network.go

```diff
@@ -158,30 +166,30 @@ func createEthClientConnection(id int64, port uint) ethclient.EthClient {
 	return ethnode
 }
 
-func deployContract(workerClient ethclient.EthClient, w wallet.Wallet, contractBytes []byte) *common.Address {
+func DeployContract(workerClient ethclient.EthClient, w wallet.Wallet, contractBytes []byte, nonce uint64) (*common.Address, error) {
```

Comment:
Note that `DeployContract` now takes a nonce and returns an error.

This is required because the contract deployer doesn't know what nonce to use, so it has to start from zero and iterate in response to errors. This isn't an issue in the sims, because the wallet is created at the start and can keep track of the correct nonce to use.